// postgres-wookie/sites.js
function sites (pgClient, opts) {
  var client = new pgClient()
  client.connect()

  // console.log(opts)

  return {
      insert: insert
    , selectAll: selectAll
    , selectSite: selectSite
  }

  function selectAll (cb) {

    client.query('SELECT * FROM sites', function (err, data) {
      cb(err, data)
    })

  }

  function selectSite (name, cb) {

    client.query('SELECT * FROM sites WHERE name = ($1)', [name])

  }

  function insert(site, cb) {

    checkExistsName(site, function (err, data) {
      console.log(data)
      if (data === false) runInsert()
      else cb(null, 'Already exists')
    })

    function runInsert() {
      console.log(site)
      if (site.name === '') cb('Site must have a name')
      if (site.created_by_user_id === '' || site.created_by_user_id === '0') cb('Invalid created_by_user_id')
      else {
        var query = ['INSERT INTO sites (name, created_by_user_id) VALUES ($1, $2)', [site.name , site.created_by_user_id]]
  
        client.query(query[0], query[1], function (err, data) {
          if (err) console.error(err)
          cb(err, data)
        })   
      }   
    }

  }

  function checkExistsName(site, cb) {
    var query = 'SELECT * FROM sites WHERE name = ($1)'
    client.query(query, [site.name], function (err, data) {
      if (err) console.error(err)
      else {
        if (data.rows.length >= 1) cb(null, true)
        else cb(null, false)
      }
    })
  }


}

module.exports = sites
